### PR TITLE
[IMP]  open_academy: computed fields and default values t#59945

### PR DESCRIPTION
**Pull request by JJRR5**
**Finished the computed fields and default values section of Building a Module Course.

The main goal of this section was to understand how to generate computed methods and how to add default values to the fields.**

- Created a computed method to calculate the percentage of taken seats to the Session model
- Created the tree and form views for the Session model and added the taken_seats field into them.
- Displayed the taken_seats field as a progress bar widget 
- Defined the start_date value of the session as the current day (today)
- Created a field to know is the session is active. 

**1.- Image Session Form View with new fields**
image

**2.- Image Session Tree View with taken seats** 
image

### CODE REVIEW

**Comment by JJRR5** [comment]
@sebasdrk17 Could you give me a hand and check that everything is ok?

File: open_academy/models/session.py [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-    start_date = fields.Date()
+    start_date = fields.Date(default=fields.Date.today())
     duration = fields.Integer()
     number_of_seats = fields.Integer()
     instructor_id = fields.Many2one('res.partner')
     course_id = fields.Many2one('course', required=True)
     attendees_ids = fields.Many2many('res.partner', string="Attendees")
+    active = fields.Boolean(default=True)
+    taken_seats = fields.Integer(compute='_compute_percentage')
+    @api.depends('number_of_seats', 'attendees_ids')
```
Comment (sebasdrk17) [inline open_academy/models/session.py:16]: add a blank space please
```suggestion

    @api.depends('number_of_seats', 'attendees_ids')
```

File: open_academy/models/session.py [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
+    def _compute_percentage(self):
```
Comment (sebasdrk17) [inline open_academy/models/session.py]: Check oca guidelines for models [here](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#models)

```suggestion
    def _compute_taken_seats(self):
```

**Review by sebasdrk17 — CHANGES_REQUESTED** [review]
Asked for some changes

**Comment by JJRR5** [comment]
@sebasdrk17 could you please check?

**Review by sebasdrk17 — APPROVED** [review]
LGTM, could you check please? @andreaarce

File: open_academy/models/session.py [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
     attendees_ids = fields.Many2many('res.partner', string="Attendees")
+    active = fields.Boolean(default=True)
+    taken_seats = fields.Integer(compute='_compute_taken_seats')
+
+    @api.depends('number_of_seats', 'attendees_ids')
+    def _compute_taken_seats(self):
+        for record in self:
+            if record.number_of_seats > 0:
+                record.taken_seats = round((len(record.attendees_ids) * 100) / record.number_of_seats)
+            else:
```
Comment (andreaarce) [inline open_academy/models/session.py]: You can skip the else if you add a value to taken seat before something like 
```suggestion
                for record in self:
                    taken_seats = 0
                        if record.number_of_seats > 0:
                            taken_seats = round((len(record.attendees_ids) * 100) / record.number_of_seats)
                        record.taken_seats = taken_seats
```

**Comment by JJRR5** [comment]
@andreaarce I changed and tested what you suggested and it works, could you please check?

**Comment by JJRR5** [comment]
Changes tested @sebasdrk17

**Review by sebasdrk17 — APPROVED** [review]

**Pull request merged**